packages/core/src/asciidoctor-core-api.js

```diff
@@ -793,6 +793,48 @@ AbstractNode.prototype.applySubstitutions = function (text, subs) {
   return this.$apply_subs(text, subs)
 }
 
+/**
+ * Resolve the list of comma-delimited subs against the possible options.
+ *
+ * @param {string} subs - The comma-delimited String of substitution names or aliases.
+ * @param {string} [type] - A String representing the context for which the subs are being resolved (default: 'block').
+ * @param {Array<string>} [defaults] - An Array of substitutions to start with when computing incremental substitutions (default: undefined).
+ * @param {string} [subject] - The String to use in log messages to communicate the subject for which subs are being resolved (default: undefined)
+ *
+ * @returns {Array<string>} - An Array of Strings representing the substitution operation or nothing if no subs are found.
+ * @memberof AbstractNode
+ */
+AbstractNode.prototype.resolveSubstitutions = function (subs, type, defaults, subject) {
+  if (typeof type === 'undefined') {
+    type = 'block'
+  }
+  if (typeof defaults === 'undefined') {
+    defaults = Opal.nil
+  }
+  if (typeof subject === 'undefined') {
+    subject = Opal.nil
+  }
```

Comment:
Would it be appropriate to also check for null?

Reply:
I'm not sure, undefined means basically, the parameter was absent. If the value is `null` then you pass explicitly a `null` value:

```js
resolveSubstitutions('attributes+') // type, defaults and subject are undefined
```
```js
resolveSubstitutions('attributes+', null, null, null) // type, defaults and subject are null
```

I think it's a mis-usage of the API so I don't really want to support it by converting `null` to `Opal.nil`.